modules/agar/src/test/ts/browser/assertions/ApproxComparisonsTest.ts

```diff
@@ -0,0 +1,129 @@
+import { context, describe, it } from '@ephox/bedrock-client';
+import { assert } from 'chai';
+
+import * as ApproxComparisons from 'ephox/agar/assertions/ApproxComparisons';
+
+describe('browser.agar.assertions.ApproxComparisonsTest', () => {
+
+  context('measurement', () => {
+    it('not a string', () => {
+      assert.throws(
+        () => ApproxComparisons.measurement(10, 'px', 0).strAssert('Label', 2020 as any),
+        '"2020" did not have the correct unit. Expected: "px", but was: "2020"'
+      );
+    });
+
+    it('not a measurement', () => {
+      assert.throws(
+        () => ApproxComparisons.measurement(10, 'px', 0).strAssert('Label', 'goose'),
+        '"goose" was not a valid measurement'
+      );
+    });
+
+    it('wrong unit', () => {
+      assert.throws(
+        () => ApproxComparisons.measurement(10, 'px', 0).strAssert('Label', '10rem'),
+        '"10rem" did not have the correct unit. Expected: "px", but was: "rem"'
+      );
+    });
+
+    it('wrong unit, wrong margin', () => {
```

Comment:
This test doesn't have anything to do with margin. Should there be a separated test for `wrong margin`?

Reply:
Testing for the correct unit came first. Hence it didn't get to compare the margin